backend/routes.js

```diff
@@ -1,17 +1,27 @@
 const express = require('express');
-
 const router = express.Router();
+const User = require('./models/userModel.js');
+const { createUser,login } = require('./controllers/userController.js');
+const bcrypt = require('bcrypt');
+const saltRounds = 12;
+
 
-// sample route
-// a  route like this would fire
-router.get('/', (req, res) => {
-    try {
-        res.json({msg: 'test GET request working'});
-    } catch(error) {
-        res.status(400).json({error: 'something went wrong'});
-    }
+router.post('/newUser', async (req, res) => {
+  try {
+    const { username, email, password } = req.body;
+    const encryptedPassword = await bcrypt.hash(password, saltRounds);
+    const newUser = new User({ username, email, password: encryptedPassword });
+    const savedUser = await newUser.save();
+    res.status(201).json(savedUser);
```

Comment:
Can you remove this code that hashes the password and saves the user in the DB and instead call the createUser function like you're doing for the login route down below, or is there a reason you included db functions in the router file?